enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -394,12 +400,19 @@ type podmanCommandContainer struct {
 	removed bool
 }
 
+// TODO: remove this and always use provider.New() instead.
 func NewPodmanCommandContainer(env environment.Env, image, buildRoot string, options *PodmanOptions) container.CommandContainer {
+	imageExistsCache, _ := newImageExistsCache()
```

Comment:
I think you wanna pass this in from the Provider so it's shared across CommandContainers, right?

Reply:
provider.New() doesn't call this func unfortunately - this func is only used for testing. Line 403 has a TODO to remove this func and only use the provider, even in tests

Reply:
Oof, got me! Maybe one of us should get on that TODO soon 😛

Reply:
+1

Reply:
I can do it :) had started on it in this PR, but then it started getting super big so I split it into a separate branch